### ISSUE TITLE
Need some help to get started

### ISSUE THREAD

**Issue by amo13** [M0Rf30/simonpi#12]
Thank you a lot for providing this very handy script!

I created an rpi4 image by calling `simonpi  rpi-4 -s 6` and got a file called `sd-arch-rpi-4-qemu` at the end. Then I launched it with QEMU (using virtual machine manager gui) by selecting this image file, telling the virtual machine that it is archlinux and by creating a 20gb virtual drive.
Unfortunately, when booting the virtual machine, I only get a message stating that no bootable device was found. 
Do you have an idea what I might be doing wrong?

**Comment by M0Rf30** [comment]
Hi @amo13 . Please run ```simonpi rpi-4 -r``` after the creation of ```sd-arch-rpi-4-qemu```.

**Comment by amo13** [comment]
Thank you for your fast reply!
I am not quite there yet, the result of `simonpi rpi-4 -r` is 
```
[  OK  ] Mapping lenghts of partitions ...
[sudo] password for amo: 
[  OK  ] Mounting disk image named sd-arch-rpi-4-qemu.img ...
[  OK  ] Checking partitions to prevent failures ...
[ WARN ] rpi-4/boot folder is present
[ WARN ] rpi-4/root folder is present
[  OK  ] Mounting partitions of sd-arch-rpi-4-qemu.img ...
[ WARN ] Out of a Docker container ...
[  OK  ] Turning up a network for QEMU ...
[ WARN ] Port 53 is busy
[ WARN ] Trying to use local dns service ( maybe offline )
[  OK  ] Gateway address: 192.168.66.1
qemu-system-aarch64: Initialization of device cfi.pflash01 failed: Could not reopen file: Permission denied
```

**Comment by M0Rf30** [comment]
Please try to clean your environment using ```simonpi rpi-4 -k && simonpi rpi-4 -u```, clean your environment with  ```simonpi rpi-4 -p```. recreate your sd-image and finally run ```simonpi rpi-4 -r``` .
If none of these instructions work for you, please attach a ```bash -x simonpi rpi-4 -r``` log execution here and I will try to fix issues.

**Comment by amo13** [comment]
Thank you a lot for your help!
I attached a log, I hope it helps figuring out what is wrong.
Another thing is that I am confused about whether to run simonpi with or without sudo. If I do without, simonpi complains about missing privileges (see attached log, line 81), and if I do with, it says `qemu-system-aarch64: failed to load "/root/.simonpi/rpi-4/boot/initramfs-linux.img"` (line 94)
[simonpi.log](https://github.com/M0Rf30/simonpi/files/4774813/simonpi.log)
Which commands am I supposed to call with sudo and which ones without?

**Comment by M0Rf30** [comment]
Ok thank you for the log. here some explainations. I will check for issues in the next days. Today is just relax time, but I want to explain some considerations. Do not execute simonpi as root by default. simonpi try to use sudo only when it's needed. So run it with no sudo at all and simonpi will ask for privilege during its process. At a first glance the issue seems to be related to the pflash image (the one under /usr/share/edk2/aarch64). try to play with it or disable it /opt/simonpiemu/scripts/runemu file. if you don't need an EFI environment this should do the trick, at least for now. remember to run chown the needed files, because of the last execution with sudo

**Comment by M0Rf30** [comment]
here an example of execution (during the -s stage I just have the Arch Linux Arm images so simonpi quickly create the sd image)
https://asciinema.org/a/E876twzRUyxDRUrFaO5uZVPEe
hope it helps

**Comment by amo13** [comment]
I appreciate your help with this, thank you!
Thank you also for clearing up the sudo thing... I could have guessed that is was meant that way :)

Doing a simple `chown -R user:user /usr/share/edk2/aarch64` did the trick, you were right with playing around with the pflash stuff. It is working fine now and I did not have to disable anything. 

Maybe the problem is related to the BUILD file of the AUR package?

Anyway, thank you. Closing this for now as I am now able to get started =)
